ngraph/core/reference/src/runtime/reference/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -0,0 +1,389 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/runtime/reference/experimental_detectron_roi_feature_extractor.hpp"
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstring>
+#include <numeric>
+
+#include "ngraph/op/experimental_detectron_roi_feature.hpp"
+#include "ngraph/shape.hpp"
+
+#if defined(__GNUC__) && !defined(__clang__)
+#    if defined(__linux__) && defined(__i386__) && (__GNUC__ == 7 && __GNUC_MINOR__ == 5 && __GNUC_PATCHLEVEL__ == 0)
+#        define NEED_FIX 1
```

Comment:
Why do we introduce this define? Can we try to fix it?

Reply:
No, we cannot. The reason of this `define` is incorrect compiler optimization of the function `pre_calc_for_bilinear_interpolate()` in GCC 7.5.0 on Ubuntu 18.04 32-bit (IA-32 processor mode). But Clang-10 (with the same optimization level) optimizes this function correctly, on the same Ubuntu 18.04 32-bit.